flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -559,6 +565,7 @@ public void onException(Throwable cause) {
 		//            the fetchers 'snapshotCurrentState()' method return at least
 		//            the restored offsets
 		this.kafkaFetcher = fetcher;
+		this.kafkaOffsetCommitter = createOffsetCommitter();
```

Comment:
Changing order of those lines:
```
		this.kafkaFetcher = fetcher;
		this.kafkaOffsetCommitter = createOffsetCommitter();
```
will brake the code. This is super fragile and is kind of hard to follow.

Reply:
This is a very valid argument. Will address this with a factory perhaps, as you suggested.